src/app/core/pages/Authcomponents/signup/signup.component.ts

```diff
@@ -0,0 +1,61 @@
+
+import { Component, inject } from '@angular/core';
+import { AbstractControl, FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { HttpErrorResponse } from '@angular/common/http';
+
+import { Router, RouterLink } from '@angular/router';
+import { Subject, takeUntil } from 'rxjs';
+import { AlertsComponent } from '../../../../shared/components/ui/alerts/alerts.component';
+import { EmailInputComponent } from '../../../../shared/components/ui/email-input/email-input.component';
+import { BackgroundComponent } from '../../../../shared/components/ui/background/background.component';
+import { IconComponent } from '../../../../shared/components/ui/icon/icon.component';
+import { RegButtonComponent } from '../../../../shared/components/ui/reg-button/reg-button.component';
+import { AuthApiService } from '../../../../../../projects/auth-api/src/public-api';
+import { validsignup } from '../../../../shared/utilites/validsignup';
+
+@Component({
+  selector: 'app-signup',
+  imports: [AlertsComponent, EmailInputComponent, ReactiveFormsModule, BackgroundComponent, IconComponent, RegButtonComponent],
+  templateUrl: './signup.component.html',
+  styleUrl: './signup.component.scss'
+})
+export class SignupComponent {
+  private ngUnsubscribe = new Subject<void>(); 
+
+  private _router = inject(Router)
+  errormessage : string="";
+  private _auth=inject(AuthApiService)
```

Comment:
```suggestion
  private _auth=inject(AuthApiService)
```
change to comment too in login